src/types/timezone.jl

```diff
@@ -68,20 +86,40 @@ function TimeZone(str::AbstractString, mask::Class=Class(:DEFAULT))
     # Note: If the class `mask` does not match the time zone we'll still load the
     # information into the cache to ensure the result is consistent.
     tz, class = get!(_tz_cache(), str) do
-        tz_path = joinpath(TZData.COMPILED_DIR, split(str, "/")...)
-
-        if isfile(tz_path)
-            open(deserialize, tz_path, "r")
-        elseif occursin(FIXED_TIME_ZONE_REGEX, str)
-            FixedTimeZone(str), Class(:FIXED)
-        elseif !isdir(TZData.COMPILED_DIR) || isempty(readdir(TZData.COMPILED_DIR))
-            # Note: Julia 1.0 supresses the build logs which can hide issues in time zone
-            # compliation which result in no tzdata time zones being available.
-            throw(ArgumentError(
-                "Unable to find time zone \"$str\". Try running `TimeZones.build()`."
-            ))
+        # Even though we're using Thread-local caches, we still need to lock during
+        # construction to prevent multiple tasks redundantly constructing the same object,
+        # and potential thread safety violations due to Tasks migrating threads.
+        # NOTE that we only grab the lock if the TZ doesn't exist, so the mutex contention
+        # is not on the critical path for most constructors. :)
+        constructing = false
+        # We lock the mutex, but for only a short, *constant time* duration, to grab the
+        # future for this TimeZone, or create the future if it doesn't exist.
+        future = @lock tz_cache_mutex begin
+            get!(TZ_CACHE_FUTURES, str) do
+                constructing = true
+                Channel{Tuple{TimeZone,Class}}(1)
+            end
+        end
+        if constructing
+            tz_path = joinpath(TZData.COMPILED_DIR, split(str, "/")...)
+
+            tz = if isfile(tz_path)
```

Comment:
Returned value is a tuple not just a timezone
```suggestion
            t = if isfile(tz_path)
```

Reply:
Done. thanks!